aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1883,164 +1778,176 @@ impl VMAdapter for AptosVM {
+            &mut session,
+            &resolver,
+            &txn,
+            &txn_data,
+            &log_context,
+        );
+
+        // Increment the counter for transactions verified.
+        let (counter_label, result) = match validation_result {
+            // For validation, having too new sequence numbers is ok.
+            Err(err) if err.status_code() != StatusCode::SEQUENCE_NUMBER_TOO_NEW => (
+                "failure",
+                VMValidatorResult::new(Some(err.status_code()), 0),
+            ),
+            _ => (
+                "success",
+                VMValidatorResult::new(None, txn.gas_unit_price()),
             ),
         };
 
-        match result {
-            Ok(output) => output,
-            Err(err) => {
-                // Invalidate the loader cache in case there was a new module loaded from a module
-                // publish request that failed.
-                // This ensures the loader cache is flushed later to align storage with the cache.
-                // None of the modules in the bundle will be committed to storage,
-                // but some of them may have ended up in the cache.
-                if new_published_modules_loaded {
-                    self.0 .0.mark_loader_cache_as_invalid();
-                };
-                let txn_status = TransactionStatus::from_vm_status(
-                    err.clone(),
-                    self.0
-                         .0
-                        .get_features()
-                        .is_enabled(FeatureFlag::CHARGE_INVARIANT_VIOLATION),
-                );
-                if txn_status.is_discarded() {
-                    discard_error_vm_status(err)
-                } else {
-                    let (vm_status, output) = self.0.failed_transaction_cleanup_and_keep_vm_status(
-                        err,
-                        &gas_meter,
-                        &txn_data,
-                        resolver,
-                        log_context,
-                        &storage_gas_params.change_set_configs,
-                    );
-                    (vm_status, output)
-                }
-            },
-        }
+        TRANSACTIONS_VALIDATED
+            .with_label_values(&[counter_label])
+            .inc();
+
+        result
+    }
+}
+
+impl VMSimulator for AptosVM {
+    fn simulate_signed_transaction(
+        transaction: SignedTransaction,
+        state_view: &(impl StateView + Sync),
+    ) -> Result<TransactionOutput, VMStatus> {
+        // The caller must ensure that the signature is not invalid, as otherwise
+        // this can be an attack.
+        let txn =
+            SignatureVerifiedTransaction::valid_for_simulation(transaction).ok_or_else(|| {
+                VMStatus::error(
+                    StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR,
+                    Some("Simulated transaction should not have a valid signature".to_string()),
+                )
+            })?;
```

Comment:
Let's change this `can be an attack` to `as otherwise a malicious actor could execute the transaction without their knowledge`

There's trust involved, where if you send the transaction to the wrong endpoint, you could just lose it regardless of whether we reject it here.

Future versions of simulation should not accept a signature at all so that isn't a possibility of confusion / attack.